### PR TITLE
Added "ggplotgui" code

### LINKED ISSUE #5146: New dialog: Graph GUI (ggplotgui)
There is a new package ggplotgui which has a nice interface for constructing graphs including getting the equivalent R code https://cran.r-project.org/web/packages/ggplotgui/README.html

This will be a very simple dialog to add as it just needs to run `ggplotgui::ggplot_shiny(df)` to launch the gui. 

It could go towards the bottom of the Describe menu as `Graph GUI`.

@maxwellfundi please assign to an appropriate person in @africanmathsinitiative/developers 

The only potential issue might be if the gui doesn't load correctly through R-Instat.

This is for the upcoming release.

### PR DESCRIPTION
**Pull request by Ogik99**
fixes #5146 
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

File: instat/dlgGraphsGUI.vb [hunk 1]
```diff
@@ -1,3 +1,91 @@
...
+' but WITHOUT ANY WARRANTY; without even the implied warranty of
+' MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+' GNU General Public License for more details.
+'
+' You should have received a copy of the GNU General Public License 
+' along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-End Class
\ No newline at end of file
+Imports instat
```
Comment (maxwellfundi) [inline instat/dlgGraphsGUI.vb]: This line can be removed. Its not needed!

File: instat/dlgGraphsGUI.vb [hunk 2]
```diff
@@ -1,3 +1,91 @@
...
+        SetRCodeForControls(True)
+        TestOkEnabled()
+    End Sub
+
+    Private Sub ucrReceiverGraphs_ControlContentsChanged(ucrChangedControl As ucrCore) Handles ucrReceiverGraphs.ControlContentsChanged
+        TestOkEnabled()
+    End Sub
+
+    Private Sub ucrSelectorDataFramesToDelete_Load(sender As Object, e As EventArgs)
+
```
Comment (maxwellfundi) [inline instat/dlgGraphsGUI.vb]: This empty sub can also be removed @Ogik99

**Comment by Ogik99** [comment]
@maxwellfundi I have made the recommended changes. Kindly review.

**Comment by dannyparsons** [comment]
@Ogik99 @Ivanluv thanks for working on this. The dialog set up is ok but the R connection has some issues of crashing while the app is running. We'll have to keep looking into this and hopefully have it working for the next released version.

**Comment by maxwellfundi** [comment]
@dannyparsons A quick reminder to have a look on this.

**Comment by dannyparsons** [comment]
Yes I couldn't see an easy way to solve the crashing issue. It might be to be in touch with rdotnet developers.

File: instat/dlgGraphsGUI.vb [hunk 3]
```diff
@@ -1,3 +1,85 @@
...
+        TestOkEnabled()
+    End Sub
+
+    Private Sub TestOkEnabled()
+        If ucrReceiverGraphs.IsEmpty Then
+            ucrBase.OKEnabled(False)
+        Else
+            ucrBase.OKEnabled(True)
+        End If
+
```
Comment (Patowhiz) [inline instat/dlgGraphsGUI.vb]: Though trivial, you could have the entire if statement as 1 line  `ucrBase.OKEnabled(Not ucrReceiverGraphs.IsEmpty)`

**Comment by Patowhiz** [comment]
@dannyparsons  So are we still going to add it to the menu and have it accessible?

**Comment by dannyparsons** [comment]
Not yet as its currently unstable, but yes once we resolve this.

**Comment by shadrackkibet** [comment]
@Ogik99 what is the state of this? Anything remaining?

**Comment by lloyddewit** [comment]
@Patowhiz Please could you try and complete this PR? @rdstern suggested this.

**Comment by Patowhiz** [comment]
@lloyddewit thank you.  Will take it up

**Comment by lloyddewit** [comment]
@Patowhiz Did you get a chance to look at this? thanks

**Comment by Patowhiz** [comment]
Yes. I'm doing this in a separate branch which I will push once my questions are answered in issue #5146. Its clear to me that this dialog needs a package that is dependent on other major packages, which they are currently not shipped with R-Instat.

**Review by rdstern — CHANGES_REQUESTED** [review]
It would be great if this could be added.  It is from Feb 2019 from @dannyparsons so could he re-allocate this?

**Comment by lloyddewit** [comment]
@rdstern @dannyparsons @Patowhiz 
I suggest that @patowhiz:
- tests that installing packages `devtools` and `ggpoltgui` do not cause any issues for R-Instat
- if ok, create new issue `Packages for next version` similar to issue #5976
- links this issue to this PR
- completes this PR

If I misunderstood anything or you have other comments then please let me know, thanks

**Comment by dannyparsons** [comment]
`devtools` is already a package included in R-Instat so that's not an issue. I think the issue is whether we can launch a shiny app from within R-Instat.

If it doesn't work, there might not be much we can do as it may be an issue with rdotnet. To test this I would suggest making a test VB repo with a single form that uses rdotnet to launch the shiny app and see if it works. e.g.

Imports RDotNet

```vb
Public Class Form1

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim engine As REngine

        REngine.SetEnvironmentVariables()
        engine = REngine.GetInstance()
        engine.Initialize()

        engine.AutoPrint = True

        engine.Evaluate("some_r_code")
    End Sub
End Class
```

If it works then we should be able to get it working in R-Instat too. If not, then we could report to the rdotnet repo but probably can't do anything ourselves.

**Comment by lloyddewit** [comment]
No work was done on this PR for nearly three years so I closed this PR. It is still available for reference if needed (this PR is referenced in issue #5146).

**Pull request closed without merging**